editor/src/clj/editor/properties.clj

```diff
@@ -491,17 +505,35 @@
 (defn read-only? [property]
   (:read-only? property))
 
+(defn user-edit?
+  "Callable from a property setter function. Returns true if the setter function
+  is being invoked as a result of the user editing the specified property. It
+  will return false if the property setter is being called as a result of other
+  operations such as resource loading, script code being executed, etc."
+  [node-id prop-kw evaluation-context]
+  {:pre [(map? evaluation-context)]}
+  (true?
+    (some-> evaluation-context
+            :tx-data-context
+            deref
+            :edited-endpoints
+            (contains? (g/endpoint node-id prop-kw)))))
+
 (defn set-values!
+  "Set values as a result of a user-edit in the Property Editor."
   ([property values]
    (set-values! property values (gensym)))
   ([property values op-seq]
    (when (not (read-only? property))
-     (let [evaluation-context (g/make-evaluation-context)]
+     (let [evaluation-context (g/make-evaluation-context)
+           set-operations (resolve-set-operations property values)
+           edited-endpoints (edited-endpoints set-operations)]
        (g/transact
+         {:tx-data-context-map {:edited-endpoints edited-endpoints}}
```

Comment:
I the future we might want to do this from the `form-view` as well, but currently only the Property Editor is used to edit the properties that call the `properties/user-edit?` function.